### ISSUE TITLE
How to remap urls to scripts

### ISSUE THREAD

**Issue by ajxchapman** [ajxchapman/ReServ#10]
The following routes can all map to the same script:
```json
[
  {
        "protocol" : "http",
        "route" : "^/(.*)",
        "path" : "./wwwroot/$1"
    },
    {
        "protocol" : "http",
        "route" : "^/scripts/(.*)",
        "path" : "./wwwroot/scripts/$1"
    },
    {
        "protocol" : "http",
        "route" : "^/runme/(.*)",
        "path" : "./wwwroot/scripts/runme.py"
    },
    {
        "protocol" : "http",
        "route" : "^/test([^/]*)",
        "path" : "./wwwroot/scripts/runme.py"
    }
]
```

How should the URL be remapped in each instance?

**Comment by ajxchapman** [comment]
I think this is best addressed by adding a `remap_url`, HTTP route configuration option with a regex to grok the remapped URL from the original url

**Comment by ajxchapman** [comment]
Currently using the following
```
    "route" : "^/httpbin/.*",
    "url" : {
      "base" : "/httpbin",
      "remap" : "/httpbin/(.*)"
    }
```

This feels like a lot of unnecessary re-definition, is there a better way?

**Comment by ajxchapman** [comment]
Settling on something more like the below when a remap is required:
```
    "route" : "^/httpbin/.*",
    "path" : {
      "path" : "./scripts/httpbin.py",
      "base" : "/httpbin"
    },
```

But still allowing the usual `path` when a remap is not needed:
```
    "route" : "^/httpbin/.*",
    "path" : "./files/wwwroot/blah"
```

**Comment by ajxchapman** [comment]
Closed out in 0afea2f46e66501a40cc31e1ad6f1a408479c22b
